my-code/src/containers/Home.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react'
+import SearchInput from '../components/searchInput';
+import MovieCard from '../components/movieCard';
+import logo from '../assets/images/logo.svg';
+import { withRouter } from 'react-router-dom';
+import EmptyStateManager from '../components/emptyStateManager';
+
+class Home extends Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      data: [],
+      searchValue: '',
+      loading: false,
+      error: '',
+      addedMovies: [],
+    }
+  }
+
+  componentDidMount() {
+    if(localStorage.getItem('addedMovies')) {
+      const retrievedData = localStorage.getItem('addedMovies');
+      const parseData = JSON.parse(retrievedData);
+      this.setState({ addedMovies: parseData})
+    }
+  }
+
+  handleChange(event) {
+    const value = event.target.value;
+    this.setState({ searchValue: value,loading: true, error: '' },
+      () => {
+        this.fetchMovies(value);
```

Comment:
Make sure to only request to the server when the user is ready. I mean, it's requesting on every `keydown` event, so there is a lot of unnecessary requests here. For instance, I've searched for `007` and the network has 3 requests, but only the last one is helpful. Did you get my point?

To solve it, there are some common approaches to do it, like [debounce and throttle](http://loopinfinito.com.br/2013/09/24/throttle-e-debounce-patterns-em-javascript/). Try to think which one fits better for this problem.

Reply:
I though I wrapped this on setTimeout :/